WcaOnRails/lib/custom_i18n_scanner.rb

```diff
@@ -0,0 +1,69 @@
+class WCAFileScanner < I18n::Tasks::Scanners::FileScanner
+  include I18n::Tasks::Scanners::RelativeKeys
+  include I18n::Tasks::Scanners::OccurrenceFromPosition
+
+  def active_record?(model)
+    # FIXME: Find a way to determine this dynamically ?
+    model != "contact"
+  end
+
+  def extract_model_attr(key)
+    model_key = key.gsub(/^(\w+)\..*$/, '\1').singularize
+    # Some dirty tricks to assign the detected field to the "correct" model
+    # For most controllers the forms modify a model which has the same name
+    # as the controller, however for some of them they modified a single different
+    # model (eg: the `admin` forms modify `person`s, and only that model).
+    model_name = case model_key
+                 when "devise" then "user"
+                 when "admin" then "person"
+                 when "oauth" then "doorkeeper/application"
+                 else model_key
+                 end
+    attribute = key.gsub(/.*\.(\w+)$/, '\1')
+    [model_name, attribute]
+  end
+
+  # @return [Array<[absolute key, Results::Occurrence]>]
+  def scan_file(path)
+    text = read_file(path)
+    retval = []
+    # First scan for inputs (or labels for inputs) that are not hidden
+    text.scan(/^\s*<%= f.(input|label|input_field) :(\w+)(?!.*hidden)(.*)%>/) do |match|
+      # 'absolute_key' returns something that we assume looks like model.(.*).attribute
+      abskey = absolute_key(".#{match.second}", path)
+      occurrence = occurrence_from_position(
+        path, text, Regexp.last_match.offset(0).first)
+      model, attribute = extract_model_attr(abskey)
+      # Mark the hint as used if we don't use custom hint
+      if !match.third.include?("hint:")
+        retval << ["simple_form.hints.#{model}.#{attribute}", occurrence]
+      end
+      # Mark the label as used if we don't use custom hint
+      if !match.third.include?("label:")
+        # Simple form can fetch its labels from activerecord.attributes,
+        # Mark it as used ... Except if the model is not an ActiveRecord ;)
+        # if not we mark the label as used!
+        retval << if active_record?(model)
+                    ["activerecord.attributes.#{model}.#{attribute}", occurrence]
+                  else
+                    ["simple_form.labels.#{model}.#{attribute}", occurrence]
+                  end
+      end
+    end
+    # Then scan for possible radio choices
+    text.scan(/^\s*<%= f.input :(\w+), as: :radio_buttons, collection: \[(.*)\](.*)%>/) do |match|
+      abskey = absolute_key(".#{match.first}", path)
+      occurrence = occurrence_from_position(
+        path, text, Regexp.last_match.offset(0).first)
+      model, attribute = extract_model_attr(abskey)
+      # Get all choices from the f.input
+      options = match.second.gsub(/[: ]/, '').split(',')
+      options.each do |o|
+        # Mark every choice as used
+        retval << ["simple_form.options.#{model}.#{attribute}.#{o}", occurrence]
+      end
+    end
```

Comment:
`match.first`, `match.second`,  `match.third` forces to look on the regular expression a couple of times to figure out their meaning. It would be nicer to use some descriptive block arguments like `do |attribute, collection|` or whatever you consider expressive here. The same goes for the block above.